spring-cloud-starter-stream-source-twitterstream/src/main/java/org/springframework/cloud/stream/app/twitterstream/source/AbstractTwitterInboundChannelAdapter.java

```diff
@@ -227,16 +228,16 @@ private void waitLinearBackoff() {
 		}
 	}
 
-	private void waitRateLimitBackoff() {
+	private void waitRateLimitBackoff(HttpStatusCodeException sce) {
 		int millis = rateLimitBackOff.get();
-		logger.warn("Rate limit error, waiting for " + millis / 1000 + " seconds before restarting");
+		logger.warn("Rate limit error, waiting for " + millis / 1000 + " seconds before restarting", sce);
```

Comment:
Well, yes. I think we don't need it for the rate limit case. It is fully clear that the reason is of:
>  420 (Enhance Your Calm)

So, good catch and let me know if you can't fix it on merge! 😄